### PR TITLE
Drop old versions from Project.toml compat constraints

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
## Summary

This PR simplifies maintenance by dropping support for older versions of dependencies that are no longer needed. This reduces the version constraint complexity and focuses support on the latest stable versions.

## Changes Made

Updated the following version constraints in Project.toml to drop old versions:

- **Adapt**: `"3, 4"` → `"4"`
- **CUDA**: `"4.1.0, 5"` → `"5"`  
- **DocStringExtensions**: `"0.8, 0.9"` → `"0.9"`
- **ForwardDiff**: `"0.10, 1"` → `"1"`
- **GPUArraysCore**: `"0.1, 0.2"` → `"0.2"`
- **LinearSolve**: `"1.15, 2, 3"` → `"3"`
- **Metal**: `"0.5, 1"` → `"1"`
- **RecursiveArrayTools**: `"2, 3"` → `"3"`
- **oneAPI**: `"1.2, 2"` → `"2"`

## Benefits

- **Simplified maintenance**: Fewer version combinations to support and test
- **Reduced complexity**: Cleaner dependency specification
- **Focus on current ecosystem**: Support latest stable versions
- **Easier debugging**: Fewer version conflicts and edge cases

## Test Plan

- [x] Package resolves successfully with updated constraints
- [x] Package compiles without errors
- [x] All version constraints are properly specified
- [ ] CI passes with new constraints
- [ ] Downgrade CI works with simplified constraints

## Breaking Change Notice

This is a breaking change as it drops support for older versions of dependencies. Users will need to update their environments to use the supported versions.

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -39,27 +39,27 @@ oneAPIExt = ["oneAPI"]
...
-CUDA = "4.1.0, 5"
+Adapt = "4"
+CUDA = "5"
 ChainRulesCore = "1"
 DiffEqBase = "6.122"
-DocStringExtensions = "0.8, 0.9"
-ForwardDiff = "0.10, 1"
-GPUArraysCore = "0.1, 0.2"
+DocStringExtensions = "0.9"
+ForwardDiff = "1"
```
Comment (ChrisRackauckas) [inline Project.toml:47]:
```suggestion
ForwardDiff = "0.10.38, 1"
```
